contrib/inventory/cloudforms.py

```diff
@@ -0,0 +1,363 @@
+#!/usr/bin/python
+# vim: set fileencoding=utf-8 :
+#
+# Copyright (C) 2016 Guido Günther <agx@sigxcpu.org>
+#
+# This script is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with it.  If not, see <http://www.gnu.org/licenses/>.
+#
+# This is loosely based on the foreman inventory script
+# -- Josh Preston <jpreston@redhat.com>
+#
+
+import argparse
+import ConfigParser
+import copy
+import os
+import re
+from time import time
+import requests
+from requests.auth import HTTPBasicAuth
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+class CloudFormsInventory(object):
+    def __init__(self):
+        """
+        Main execution path
+        """
+        self.inventory = dict()  # A list of groups and the hosts in that group
+        self.cache = dict()      # Details about hosts in the inventory
+        self.params = dict()     # Params of each host
+
+        # Parse CLI arguments
+        self.parse_cli_args()
+
+        # Debug?
+        self.debug = self.args.debug
+
+        # Read settings
+        self.read_settings()
+
+        # Cache
+        if self.args.refresh_cache or not self.is_cache_valid():
+            self.update_cache()
+        else:
+            self.load_inventory_from_cache()
+            self.load_params_from_cache()
+            self.load_cache_from_cache()
+
+        data_to_print = ""
+
+        # Data to print
+        if self.args.host:
+            if self.debug:
+                print "Fetching host [%s]" % self.args.host
+            data_to_print += self.get_host_info(self.args.host)
+        else:
+            self.inventory['_meta'] = {'hostvars': {}}
+            for hostname in self.cache:
+                self.inventory['_meta']['hostvars'][hostname] = {
+                    'cloudforms': self.cache[hostname],
+                    'cloudforms_params': self.params[hostname],
```

Comment:
Why the two dicts? Seems like some stuff is duplicated between cloudforms and cloudforms_params- would it be more user-friendly to squash to a single "cloudforms_facts" dict or something?